src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputColumnLimit.java

```diff
@@ -50,4 +51,54 @@ int test1(int badFormat1, int badFormat2, final int badFormat3) throws java.lang
 
   // Very long url with valid href: href    = "www.google.com/search?hl=en&q=java+style+guide+checkstyle+check+href+length+limit&btnG=Google+Search"
   int validHrefWithWhiteSpaces = 54;
+
+  @SuppressWarnings({"LineLength"})
+  int aaaarealllllllllllllllllyyyyyyyyyyylllllllloooooooooooooooonnnnnnnnnnnnnnnnnggggggggggvvvvvvaarriaaabllee = 99; // ok, suppression used, check name in camelcase
+
+  @SuppressWarnings({"linelength"})
+  int aaaarealllllllllllllllllyyyyyyyyyyylllllllloooooooooooooooonnnnnnnnnnnnnnnnnggggggggggvvvvvvaarriaaabllee2 = 99; // ok, suppression used, check name in lowercase
+
+  @SuppressWarnings({"linelength"})
+  void longggggggggggggggggggggmethoooooooooooooooooooooodddddddddddddddddddddddddddddddddddddd(int x, int y) {} // ok
+
+  @SuppressWarnings({"linelength"})
+  void testingParametersNames(int areallllllyyyyyyyyyyyyyyyyyylonnnnggggggggggname, int anotherlongnameblahblahblah) {} // ok
+
+  @SuppressWarnings({"linelength"})
+  InputColumnLimit inputColumnLimit = (InputColumnLimit) ((Object) new com.google.checkstyle.test.chapter4formatting.rule44columnlimit.InputColumnLimit()); // ok
+
+  @SuppressWarnings({"linelength"})
+  void testing() {
+    try {
+      throwExceptionBasedOnCondition();
+
+    } catch (IOException | NullPointerException | ArrayIndexOutOfBoundsException | ClassNotFoundException ex) { // ok, got suppressed
```

Comment:
This is not a good excuse, non of identifiers are long.
All of this is easy to wrap. To test it, just run formatter it will do a job.

We need cases that are not wrap-able at all.
As you start to make InputFormatted files in chapter 4, it will be more clear.
Let's move all this updates to new Input, that we be name with prefix InputFormatted, and CI will demand auto formating of it.

Reply:
Please add cases where auto formatter does not wrap.

Reply:
Suppression like this suppress all in method that is very broad suppression we need more fine grained suppression for a line or statement that cannot be wrapped.

Reply:
> Please add cases where auto formatter does not wrap.

such cases are only long names like:

https://github.com/checkstyle/checkstyle/blob/3adc3090e37e6fd20d1f285b612b0361ad3cef43/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputColumnLimit.java#L98-L103

other than this type of cases, formatter formats the code and make it under column limit.

Reply:
> Suppression like this suppress all in method that is very broad suppression we need more fine grained suppression for a line or statement that cannot be wrapped.

can you suggest some other more reliable suppression?

Reply:
investigate https://checkstyle.org/filters/suppresswithnearbycommentfilter.html#SuppressWithNearbyCommentFilter

https://github.com/checkstyle/checkstyle/blob/b69a73234fb18ba0ef9fb78aa07b0afd0196779a/src/main/resources/google_checks.xml#L377-L383

Reply:
I found [SuppressWithNearbyTextFilter](https://checkstyle.org/filters/suppresswithnearbytextfilter.html#SuppressWithNearbyTextFilter) which I think can fulfill our requirements. This suppression works with non-treewalker modules so we dont need to worry about it.

It suppresses check with near by comment by matching the comment with specified regex. Example config:

```xml
  <module name="SuppressWithNearbyTextFilter">
    <property name="checkPattern" value="LineLength"/>
    <property name="nearbyTextPattern" value="Ignore LineLength Check"/>
  </module>
```

This config will only suppress LineLength violations. It will ignore lines which will have `// Ignore LineLength Check` comments for LineLength module. For example:

```java
/** Testing. */
public class TestingNewSuppression {
  // below line of code is proof to show that only LineLength module gets suppressed by the comment,
  // other modules will still give violations.
  int x = 10; // IGNORE LINELENGTH CHECK
  int reallllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyylongggggggggggggggggggggggname = 99; // IGNORE LINELENGTH CHECK
  int reallllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyylongggggggggggggggggggggggname2 = 99; // ignore linelength check
  // violation above cuz the comment is lower cased i.e does not match the specified message.
}
```

```
$ java -jar .\checkstyle-10.17.0-all.jar -c .\google_checks.xml .\TestingNewSuppression.java
Starting audit...
[WARN] C:\checkstyle testing\TestingGoogleStyle\.\TestingNewSuppression.java:5:7: Member name 'x' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'. [MemberName]
[WARN] C:\checkstyle testing\TestingGoogleStyle\.\TestingNewSuppression.java:7: Line is longer than 100 characters (found 170). [LineLength]
Audit done.
```

Reply:
>But it would be pain for user to every time use on/off comments just to suppress warnings for this check.

We need elegant way of suppression that is not polluting code. I am open for suggestions.

Reply:
`int reallllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyylongggggggggggggggggggggggname = 99; // IGNORE LINELENGTH CHECK` too long to already let ng , we contribute more to already existing problem.
Lets put comment above and use influence.

Reply:
SuppressionCommentFilter, SuppressWithNearbyCommentFilter 
can not be used as it is Treewalker level but LineLength is Checker Check.

So our choice is only:
SuppressWithPlainTextCommentFilter
SuppressWithNearbyTextFilter

Reply:
I think this suppression is very powerful https://checkstyle.org/filters/suppresswithnearbytextfilter.html#Example8-code

User can use it for any Check and place comment at any line he wants, it could be right above violation line or over block or over method.

Reply:
done:

https://github.com/checkstyle/checkstyle/blob/9e1cbe72334b7186f47abb33b2d86cc6ae4dfe98/src/main/resources/google_checks.xml#L22-L27

https://github.com/checkstyle/checkstyle/blob/9e1cbe72334b7186f47abb33b2d86cc6ae4dfe98/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputColumnLimit.java#L55-L63

intentionally kept the wrap-able examples to show that users can misuse this suppression which they should not do

Reply:
also updated google_style.xml's suppression section because new suppression was added